core/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala

```diff
@@ -345,10 +345,26 @@ class CollectorServiceSpec extends Specification {
         e.userAgent shouldEqual "ua"
         e.refererUri shouldEqual "ref"
         e.hostname shouldEqual "h"
-        e.networkUserId shouldEqual "nuid"
+        e.networkUserId shouldEqual ""
         e.headers shouldEqual (List(l) ++ ct).map(_.toString).asJava
         e.contentType shouldEqual ct.get
       }
+      "have an empty nuid if SP-Anonymous is present" in {
+        val l  = `Location`("l")
+        val ct = Some("image/gif")
+        val r  = HttpRequest().withHeaders(l :: hs)
+        val e =
+          service.buildEvent(None, Some("b"), "p", Some("ua"), Some("ref"), "h", "unknown", r, "nuid", ct, Some("*"))
+        e.networkUserId shouldEqual ""
+      }
```

Comment:
You might want to add a test where you add a cookie with a `network_id` in the `HttpRequest`, it seems that you can do it like [that](https://stackoverflow.com/a/43746865/8568080).

Reply:
Managed to add 2 tests at the service level that sets the nuid cookie. The HttpRequest has already been deconstructed by the time we get to the service so I had to split the HttpRequest and HttpCookie into separate params to test it. Perhaps theres a better way(?) but this was the best option I could find whilst keeping the units seperate.

Reply:
:+1: